Doesn't change it according to 'tc -d qdisc'

---

--- a/net/sched/Kconfig
+++ b/net/sched/Kconfig
@@ -432,6 +432,9 @@ choice
 	config DEFAULT_FQ
 		bool "Fair Queue" if NET_SCH_FQ
 
+	config DEFAULT_CAKE
+		bool "Common Applications Kept Enhanced" if NET_SCH_CAKE
+
 	config DEFAULT_CODEL
 		bool "Controlled Delay" if NET_SCH_CODEL
 
@@ -449,6 +452,7 @@ config DEFAULT_NET_SCH
 	string
 	default "pfifo_fast" if DEFAULT_PFIFO_FAST
 	default "fq" if DEFAULT_FQ
+	default "cake" if DEFAULT_CAKE
 	default "fq_codel" if DEFAULT_FQ_CODEL
 	default "sfq" if DEFAULT_SFQ
 	default "pfifo_fast"
