diff --git a/kernel/sched/bmq.c b/kernel/sched/bmq.c
index e4a8da074702d929225e9ebcf72efa273c90232f..5db8d6d50c49362e05d261bfbfc0cca2f3b40ccd 100644
--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -2784,6 +2784,17 @@ unsigned long long task_sched_runtime(struct task_struct *p)
 	return ns;
 }
 
+DEFINE_PER_CPU(unsigned long, thermal_pressure);
+
+void arch_set_thermal_pressure(struct cpumask *cpus,
+			       unsigned long th_pressure)
+{
+	int cpu;
+
+	for_each_cpu(cpu, cpus)
+		WRITE_ONCE(per_cpu(thermal_pressure, cpu), th_pressure);
+}
+
 /* This manages tasks that have run out of timeslice during a scheduler_tick */
 static inline void scheduler_task_tick(struct rq *rq)
 {



 
