https://github.com/ckolivas/linux/pull/17/commits/3d6e52f1415a5bcdc6b06f10f9a48388597fcd60

---

--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -109,6 +109,18 @@
 
 #define RESCHED_US	(100) /* Reschedule if less than this many μs left */
 
+/*
+ * This allows printing both to /proc/sched_debug and
+ * to the console
+ */
+#define SEQ_printf(m, x...)			\
+ do {						\
+	if (m)					\
+		seq_printf(m, x);		\
+	else					\
+		pr_cont(x);			\
+ } while (0)
+
 void print_scheduler_version(void)
 {
 	printk(KERN_INFO "MuQSS CPU scheduler v0.193 by Con Kolivas.\n");
@@ -7298,7 +7310,11 @@
 
 void proc_sched_show_task(struct task_struct *p, struct pid_namespace *ns,
 			  struct seq_file *m)
-{}
+
+{
+	SEQ_printf(m, "%s (%d, #threads: %d)\n", p->comm, task_pid_nr_ns(p, ns),
+						get_nr_threads(p));
+}
 
 void proc_sched_set_task(struct task_struct *p)
 {}
