https://github.com/zen-kernel/zen-kernel/issues/132

---

--- a/kernel/sched/MuQSS.c
+++ b/kernel/sched/MuQSS.c
@@ -1011,6 +1011,10 @@ static void resched_curr(struct rq *rq)
 #define CPUIDLE_THREAD_BUSY	(16)
 #define CPUIDLE_DIFF_NODE	(32)
 
+#ifdef CONFIG_SCHED_SMT
+DEFINE_STATIC_KEY_FALSE(sched_smt_present);
+#endif
+
 /*
  * The best idle CPU is chosen according to the CPUIDLE ranking above where the
  * lowest value would give the most suitable CPU to schedule p onto next. The
