From 1807bc03b89cfecc232aa44eb3bacc985720537c Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sat, 12 Aug 2017 12:02:04 +1000
Subject: [PATCH 15/17] Swap sucks.

---
 mm/vmscan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 910e02c793ff..b55e217a15ba 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -164,7 +164,7 @@ struct scan_control {
 /*
  * From 0 .. 100.  Higher means more swappy.
  */
-int vm_swappiness = 60;
+int vm_swappiness = 44;
 /*
  * The total number of pages which are beyond the high watermark within all
  * zones.
-- 
2.19.1

