From f32edf09630976db824dd591d012d4579f3c6010 Mon Sep 17 00:00:00 2001
From: Jarmo Tiitto <jarmo.tiitto@gmail.com>
Date: Thu, 1 Jun 2017 16:38:13 +0300
Subject: [PATCH 09/21] Fixup buildGCOVKernel, exit status never worked propely
 before.

---
 buildGCOVKernel | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/buildGCOVKernel b/buildGCOVKernel
index ac26016785c4..eaf0f0ae7b9a 100755
--- a/buildGCOVKernel
+++ b/buildGCOVKernel
@@ -28,8 +28,12 @@ echo 'Building Instrumented Kernel.'
 #   gcc produces invalid code producing 'TLS' references to the gcov variables but kernel-code does not have TLS.
 # Also we are thinking the kernel as one monolith 'program' we are instrumenting.
 make $@  V=2 CC=$KGCCVER/kgcc AR=$KGCCVER/kgcc-ar NM=$KGCCVER/kgcc-nm
-[ $? -eq 0 ] && echo 'Kernel build okay!' || echo 'Kernel build failed. Stop.' && exit 1;
-
-echo 'Collecting GCNO profile files..'
-./gathergcno.sh gcnoprofile.tar.gz
-exit 0
\ No newline at end of file
+if [ $? -eq 0 ]; then
+    echo 'Kernel build okay!'
+    echo 'Collecting GCNO profile files..'
+    ./gathergcno.sh gcnoprofile.tar.gz
+exit 0
+else
+    echo 'Kernel build failed. Stop.'
+    exit 1;
+fi
-- 
2.19.2

