From b993c052245ff6525ee6701f9bc23f68210c4939 Mon Sep 17 00:00:00 2001
From: Jarmo Tiitto <jarmo.tiitto@gmail.com>
Date: Sun, 22 Apr 2018 02:10:41 +0300
Subject: [PATCH 13/21] Typo and GCC version fixups.

---
 buildGCOVKernel | 7 +++----
 calcsum-4.9.cpp | 6 +++---
 genPGOKernel    | 4 ++--
 3 files changed, 8 insertions(+), 9 deletions(-)

diff --git a/buildGCOVKernel b/buildGCOVKernel
index 6d0c08598c11..2d50ae9a3855 100755
--- a/buildGCOVKernel
+++ b/buildGCOVKernel
@@ -1,9 +1,9 @@
 #!/bin/bash
 # This script builds an GCOV instumented kernel.
 echo 'Setup instrumented kernel build env'
-KGCCVER=/opt/kgcc-7.0/bin/
-# We need to mount the kernel source in /tmp root to make the profile data path/user agnostic
-# without it gcc fails to consume the generated profile data.
+KGCCVER=/opt/kgcc-7.2/bin/
+# We need to mount the kernel source in /tmp root to make the profile data system agnostic
+# without it the generated profile data is useless.
 DIR=$PWD
 mkdir -p /tmp/kernelsrc >/dev/null 2>&1
 [ -f /tmp/kernelsrc/buildGCOVKernel ] || sudo mount --rbind $PWD /tmp/kernelsrc
@@ -30,7 +30,6 @@ echo 'Building Instrumented Kernel.'
 make $@  V=2 CC=$KGCCVER/kgcc AR=$KGCCVER/kgcc-ar NM=$KGCCVER/kgcc-nm
 if [ $? -eq 0 ]; then
     echo 'Kernel build okay!'
-    echo 'Collecting GCNO profile files..'
     ./gathergcno.sh gcnoprofile.tar.gz
     exit 0
 else
diff --git a/calcsum-4.9.cpp b/calcsum-4.9.cpp
index 3c50fb09a69f..cfacb0984bbb 100644
--- a/calcsum-4.9.cpp
+++ b/calcsum-4.9.cpp
@@ -165,12 +165,12 @@ int write_file(const vector<char> in, const char *filename)
 		fprintf(stderr, "[!] Fail to write: %s\n", filename);
 		return 2;
 	}
-	// set the gcov version in format "603*" aka gcc 6.3.*
+	// set the gcov version in format "702*" aka gcc 7.2.*
 	// the magic value is simply "*306" string casted to uint32_t
 	// or picked from the gcc error message.
-	// I have not yet figured out where kernel registers the
+	// I have not yet figured out where kgcov registers the
 	// GCC version or how to change it automatically per build.
-	unsigned int gcov_version = 1094136106;
+	unsigned int gcov_version = 1094136362;
 	fwrite(in.data(), 1, 4, fp);
 	fwrite(&gcov_version, 1, 4, fp);
 	fwrite(in.data()+8, 1, 4, fp);
diff --git a/genPGOKernel b/genPGOKernel
index 05f41bba3012..cda87a094284 100755
--- a/genPGOKernel
+++ b/genPGOKernel
@@ -3,7 +3,7 @@
 # and mangle it for consumption by PGO build
 # the kernel is built after this.
 
-KGCCVER=/opt/kgcc-7.0/bin
+KGCCVER=/opt/kgcc-7.2/bin
 echo 'Initializing kernel PGO optimization..'
 DIR=$PWD
 mkdir -p /tmp/kernelsrc
@@ -63,7 +63,7 @@ if [ ! -f $CHECKPROF/mm/.tmp_mmap.gcda ]; then
     find /tmp/kernelPGOdump/* -name '*.gcda' -print0 | xargs -0 -n1 -P0 ./rename.sh
 
 else
-    echo 'NOTE: Using previously configured profile data.'
+    echo 'NOTE: Using alredy configured profile data.'
 fi
 
 echo 'Binding profile data dir..'
-- 
2.19.2

