--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@
 VERSION = 5
 PATCHLEVEL = 3
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION = -ck1
 NAME = Bobtail Squid
 
 # *DOCUMENTATION*
@@ -15,6 +15,10 @@ NAME = Bobtail Squid
 PHONY := _all
 _all:
 
+# CKVERSION = -ck1
+# CKNAME = MuQSS Powered
+# EXTRAVERSION := $(EXTRAVERSION)$(CKVERSION)
+
 # We are using a recursive build, so we need to do a little thinking
 # to get the ordering right.
 #
