From da178919d63ecfec2738877abae02cd2ce8aa29c Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sun, 11 Dec 2016 17:08:47 +1100
Subject: [PATCH 16/16] Add -ck1 version.

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 69fa5c0310d8..1ddd9dd25e9c 100644
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,10 @@ NAME = "People's Front"
 PHONY := _all
 _all:
 
+CKVERSION = -ck1
+CKNAME = MuQSS Powered
+EXTRAVERSION := $(EXTRAVERSION)$(CKVERSION)
+
 # o Do not use make's built-in rules and variables
 #   (this increases performance and avoids hard-to-debug behaviour);
 # o Look for make include files relative to root of kernel src
-- 
2.17.1
