From 5b6cd7cfe6cf6e1263b0a5d2ee461c8058b76213 Mon Sep 17 00:00:00 2001
From: Con Kolivas <kernel@kolivas.org>
Date: Sun, 11 Dec 2016 17:08:47 +1100
Subject: [PATCH 16/16] Add -ck1 version.

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index d4d36c61940b..4a9dfe471f1f 100644
--- a/Makefile
+++ b/Makefile
@@ -15,6 +15,10 @@ NAME = Kleptomaniac Octopus
 PHONY := _all
 _all:
 
+CKVERSION = -ck1
+CKNAME = MuQSS Powered
+EXTRAVERSION := $(EXTRAVERSION)$(CKVERSION)
+
 # We are using a recursive build, so we need to do a little thinking
 # to get the ordering right.
 #
-- 
2.20.1

